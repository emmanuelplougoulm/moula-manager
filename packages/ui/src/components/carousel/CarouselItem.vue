<template>
  <div class="container">
    <div class="slides-container">
      <div v-for="(slide, index) in slides" :key="`slide-${index}`">
        <component :is="slide.svg" class="slide-image" />
        <h2 class="slide-title">
          {{ slide.title }}
        </h2>
        <div class="slide-description text-default-regular">
          {{ slide.description }}
        </div>
      </div>
    </div>
    <div class="dots-container">
      <div
        v-for="(slide, index) in slides"
        :key="`dot-${index}`"
        class="dot"
        :data-testid="`dot-${index}`"
        :class="getActiveClass(index)"
        @click="goToSlide(index)"
      ></div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue';
import type { SlideType } from './type';

function getActiveClass(index: number) {
  return currentIndex.value === index ? 'active' : '';
}

const props = defineProps({
  slides: {
    type: Array as () => SlideType[],
    required: true,
    default: () => []
  }
});

const transform = computed(() => {
  return `translateX(-${(currentIndex.value * 100) / props.slides.length}%)`;
});

const currentIndex = ref(0);
const goToSlide = (index: number) => {
  currentIndex.value = index;
};
</script>

<style scoped>
.container {
  position: relative;
  overflow: hidden;
  max-width: 21.875rem;
}

.slides-container {
  display: inline-flex;
  flex-shrink: 0;
  transition: transform 0.5s ease-in-out;
  transform: v-bind(transform);
}

.slide-image {
  width: 21.875rem;
  height: 21.875rem;
  border-radius: 0.3125rem;
}

:is(.slide-title, .slide-description) {
  text-align: center;
}

.slide-title {
  margin: 1.5rem 0 0 0;
}

.slide-description {
  line-height: 1.2rem;
  margin: 0.625rem 0 3.0313rem 0;
  color: var(--neutral-30);
  padding: 0 1.5rem;
}

.dots-container {
  width: 100%;
  display: flex;
  justify-content: center;
  gap: 0.5rem;
}

.dot {
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background-color: var(--neutral-70);
  cursor: pointer;
}

.dot.active {
  background-color: var(--primary-30);
}
</style>
