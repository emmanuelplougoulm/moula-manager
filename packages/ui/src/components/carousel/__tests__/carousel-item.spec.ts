import { mount } from '@vue/test-utils';
import CarouselItem from '../CarouselItem.vue';

describe('Carousel item tests', () => {
  it('should display 3 slides and switch images when clicking on dots', async () => {
    const wrapper = mount(CarouselItem, {
      props: {
        slides: [
          {
            svg: 'doc_checking',
            title: 'Vérifions votre association',
            description:
              'Conformément aux législations en vigueur, nous devons vérifier permettre d’utiliser nos services'
          },
          {
            svg: 'edit',
            title: 'Vos informations',
            description:
              'Conformément aux législations en vigueur, de votre association pour vous permettre d’utiliser nos services'
          },
          {
            svg: 'more_checks',
            title: 'Représentant légal',
            description:
              'Nous aurons besoin d’une pièce d’identité et de l’association pour réaliser un selfie'
          }
        ]
      }
    });
    const firstDot = wrapper.find('.dot[data-testid="dot-0"]');
    expect(firstDot.classes('active')).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();

    const secondDot = wrapper.find('.dot[data-testid="dot-1"]');
    await secondDot.trigger('click');
    expect(wrapper.html()).toMatchSnapshot();

    const thirdDot = wrapper.find('.dot[data-testid="dot-2"]');
    await thirdDot.trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
