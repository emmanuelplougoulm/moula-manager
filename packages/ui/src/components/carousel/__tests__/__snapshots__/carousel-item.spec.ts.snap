// Vitest Snapshot v1

exports[`Carousel item tests > should display 3 slides and switch images when clicking on dots 1`] = `
"<div data-v-33b50c4b=\\"\\" class=\\"container\\">
  <div data-v-33b50c4b=\\"\\" class=\\"slides-container\\">
    <div data-v-33b50c4b=\\"\\">
      <doc_checking data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></doc_checking>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vérifions votre association</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, nous devons vérifier permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <edit data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></edit>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vos informations</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, de votre association pour vous permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <more_checks data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></more_checks>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Représentant légal</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Nous aurons besoin d’une pièce d’identité et de l’association pour réaliser un selfie</div>
    </div>
  </div>
  <div data-v-33b50c4b=\\"\\" class=\\"dots-container\\">
    <div data-v-33b50c4b=\\"\\" class=\\"dot active\\" data-testid=\\"dot-0\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-1\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-2\\"></div>
  </div>
</div>"
`;

exports[`Carousel item tests > should display 3 slides and switch images when clicking on dots 2`] = `
"<div data-v-33b50c4b=\\"\\" class=\\"container\\">
  <div data-v-33b50c4b=\\"\\" class=\\"slides-container\\">
    <div data-v-33b50c4b=\\"\\">
      <doc_checking data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></doc_checking>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vérifions votre association</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, nous devons vérifier permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <edit data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></edit>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vos informations</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, de votre association pour vous permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <more_checks data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></more_checks>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Représentant légal</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Nous aurons besoin d’une pièce d’identité et de l’association pour réaliser un selfie</div>
    </div>
  </div>
  <div data-v-33b50c4b=\\"\\" class=\\"dots-container\\">
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-0\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot active\\" data-testid=\\"dot-1\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-2\\"></div>
  </div>
</div>"
`;

exports[`Carousel item tests > should display 3 slides and switch images when clicking on dots 3`] = `
"<div data-v-33b50c4b=\\"\\" class=\\"container\\">
  <div data-v-33b50c4b=\\"\\" class=\\"slides-container\\">
    <div data-v-33b50c4b=\\"\\">
      <doc_checking data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></doc_checking>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vérifions votre association</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, nous devons vérifier permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <edit data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></edit>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Vos informations</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Conformément aux législations en vigueur, de votre association pour vous permettre d’utiliser nos services</div>
    </div>
    <div data-v-33b50c4b=\\"\\">
      <more_checks data-v-33b50c4b=\\"\\" class=\\"slide-image\\"></more_checks>
      <h2 data-v-33b50c4b=\\"\\" class=\\"slide-title\\">Représentant légal</h2>
      <div data-v-33b50c4b=\\"\\" class=\\"slide-description text-default-regular\\">Nous aurons besoin d’une pièce d’identité et de l’association pour réaliser un selfie</div>
    </div>
  </div>
  <div data-v-33b50c4b=\\"\\" class=\\"dots-container\\">
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-0\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot\\" data-testid=\\"dot-1\\"></div>
    <div data-v-33b50c4b=\\"\\" class=\\"dot active\\" data-testid=\\"dot-2\\"></div>
  </div>
</div>"
`;
